File: 2026-08/src/day_29/prms_43165/Solution.java
package day_29.prms_43165;

import java.util.*;

class Solution {
    public int solution(int[] numbers, int target) {
        Queue<Integer> q = new ArrayDeque<>();
        q.offer(0);

        for (int x : numbers) {
            int sz = q.size();
            while (sz-- > 0) {
                int cur = q.poll();
                q.offer(cur + x);
                q.offer(cur - x);
            }
        }

        int cnt = 0;
        while (!q.isEmpty()) {
            if (q.poll() == target) cnt++;
        }

        return cnt;
    }
}

File: 2026-08/src/day_29/prms_43165/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

int solution(vector<int> numbers, int target) {
    queue<int> q;
    q.push(0);

    for (int x : numbers) {
        int sz = q.size();
        while (sz--) {
            int cur = q.front();
            q.pop();

            q.push(cur + x);
            q.push(cur - x);
        }
    }

    int cnt = 0;
    while (!q.empty()) {
        if (q.front() == target) cnt++;
        q.pop();
    }

    return cnt;
}
